packages/contracts/test/diamond/facets/BondingCurveFacet.t.sol

```diff
@@ -0,0 +1,217 @@
+        vm.expectEmit(true, false, false, true);
+        emit ParamsSet(connectorWeight, baseY);
+
+        vm.prank(admin);
+        IBondingCurveFacet.setParams(
+            connectorWeight,
+            baseY
+        );
+
+        assertEq(connectorWeight, IBondingCurveFacet.connectorWeight());
+        assertEq(baseY, IBondingCurveFacet.baseY());
+    }
+
+    function testSetParamsShouldRevertNotAdmin() public {
+        uint32 connWeight;
+        uint256 base;
+        uint32 connectorWeight = uint32(bound(connWeight, 1, 1000000));
+        uint256 baseY = bound(base, 1, 1000000);
+        
+        vm.expectRevert("Manager: Caller is not admin");
+        vm.prank(secondAccount);
+        IBondingCurveFacet.setParams(
+            connectorWeight,
+            baseY
+        ); 
+    }
+
+    function testDeposit(uint32 connectorWeight, uint256 baseY) public {
+        uint256 collateralDeposited;
+        uint connWeight;
+        connectorWeight = uint32(bound(connWeight, 1, 1000000));
+        baseY = bound(baseY, 1, 1000000);
+        uint256 tokenIds;
+
+        vm.expectEmit(true, false, false, true);
+        emit Deposit(secondAccount, collateralDeposited);
+
+        vm.prank(admin);
+        IBondingCurveFacet.setParams(
+            connectorWeight,
+            baseY
+        ); 
+
+        uint256 initBal = IDollar.balanceOf(secondAccount);
+
+        IBondingCurveFacet.deposit(
+            collateralDeposited, 
+            secondAccount
+        );
+
+        uint256 finBal = IDollar.balanceOf(secondAccount);
+
+        uint256 tokReturned = IBondingCurveFacet.purchaseTargetAmountFromZero(
+            collateralDeposited,
+            connectorWeight,
+            ACCURACY,
+            baseY
+        );
+
+        // Logic Test
+        uint256 baseN = collateralDeposited.add(baseY);
+        uint256 power = (baseN.mul(10 ** 18)).div(baseY);
+        uint256 result = ACCURACY.mul(SafeMath.sub((power**(connectorWeight)), 10 ** 18)).div(10 ** 18);
+
+        assertEq(collateralDeposited, IBondingCurveFacet.poolBalance());
+        assertEq(collateralDeposited, finBal - initBal);
+        assertEq(tokReturned, result);
+        assertEq(tokReturned, IBondingCurveFacet.getShare(secondAccount));
+        assertEq(tokReturned, IUbiquityNFT.balanceOf(secondAccount, 1));
+
+        uint256 newDeposit;
+    }
+
+    function testWithdraw(uint32 connectorWeight, uint256 baseY) public {
+
+        uint256 collateralDeposited;
+        uint connWeight;
+        connectorWeight = uint32(bound(connWeight, 1, 1000000));
+        baseY = bound(baseY, 1, 1000000);
+
+
+        vm.startPrank(admin);
+        IBondingCurveFacet.setParams(
+            connectorWeight,
+            baseY
+        ); 
+
+        IBondingCurveFacet.deposit(
+            collateralDeposited, 
+            secondAccount
+        );
+
+        uint256 _amount = bound(baseY, 0, collateralDeposited);
+
+        vm.expectEmit(true, false, false, true);
```

Comment:
What does `true, false, false, true` represent?

Reply:
This explains it nicely 

https://book.getfoundry.sh/cheatcodes/expect-emit